pkg/secretstorage/awsstorage/aws.go

```diff
@@ -68,27 +72,28 @@ func (s *AwsSecretStorage) Initialize(ctx context.Context) error {
 }
 
 func (s *AwsSecretStorage) Store(ctx context.Context, id secretstorage.SecretID, data []byte) error {
-	dbgLog := lg(ctx).V(logs.DebugLevel).WithValues("secretID", id)
+	msgLog := lg(ctx).WithValues("secretID", id)
 
-	dbgLog.Info("storing data")
+	msgLog.V(logs.DebugLevel).Info("storing data")
 
 	secretName := s.generateAwsSecretName(&id)
 
-	dbgLog = dbgLog.WithValues("secretname", secretName)
-	ctx = log.IntoContext(ctx, dbgLog)
+	msgLog = msgLog.WithValues("secretname", secretName)
+	ctx = log.IntoContext(ctx, msgLog)
 
 	errCreate := s.createOrUpdateAwsSecret(ctx, secretName, data)
 	if errCreate != nil {
-		return fmt.Errorf("failed to create the secret: %w", errCreate)
+		msgLog.Info("secret creation failed", "id", id, "err", errCreate.Error())
+		return errASWSecretCreationFailed
 	}
 	return nil
 }
 
 func (s *AwsSecretStorage) Get(ctx context.Context, id secretstorage.SecretID) ([]byte, error) {
-	dbgLog := lg(ctx).V(logs.DebugLevel).WithValues("secretID", id)
```

Comment:
Errors are logged regardless of the verbosity level, while info is only emitted if the corresponding verbosity level allows it. So changing this to not be a debug log and logging errors using info has IMHO the exact reverse effect to what we want. The original behavior was IMHO correct. The only change needed here was to supply our own errors instead of the ones obtained from AWS, but still keep using the `Error()` so that the errors are logged on any verbosity level.